utilities/pipelines/staticValidation/compliance/module.tests.ps1

```diff
@@ -1174,6 +1173,72 @@ Describe 'Module tests' -Tag 'Module' {
                 $outputs | Should -Contain 'systemAssignedMIPrincipalId'
             }
         }
+
+        Context 'UDT-spcific' {
+
+            It '[<moduleFolderName>] A UDT should not be of type array, but instead the parameter that uses it. AVM-Spec-Ref: BCPNFR18.' -TestCases $moduleFolderTestCases -Tag 'UDT' {
+
+                param(
+                    [hashtable] $templateFileContent
+                )
+
+                if (-not $templateFileContent.definitions) {
+                    Set-ItResult -Skipped -Because 'the module template has no user-defined types.'
+                    return
+                }
+
+                $incorrectTypes = [System.Collections.ArrayList]@()
+                foreach ($type in $templateFileContent.definitions.Keys) {
+                    if ($templateFileContent.definitions.$type.type -eq 'array') {
+                        $incorrectTypes += $type
+                    }
+                }
+
+                $incorrectTypes | Should -BeNullOrEmpty -Because ('no user-defined type should be declared as an array, but instead the parameter that uses the type. This makes the template and its parameters easier to understand. Found incorrect items: [{0}].' -f ($incorrectTypes -join ', '))
```

Comment:
Change tests to create warning instead of failing for the first version - then pivot to failing again for the final version

Reply:
done